scripts/util/resolve-types.js

```diff
@@ -157,6 +157,8 @@ module.exports = (shapesWithoutNames, { inputShapes, outputShapes }) => {
       members: Object.keys(sh.members).map(key => ({
         required: sh.required && sh.required.indexOf(key) !== -1,
         key: safeIdentifier(lowCam(key)),
+        rawKey: key,
+        decodeKeys: Array.from(new Set([key, lowCam(key), upCam(key)])),
```

Comment:
`rawKey` is used to preserve the member name, for use with json and query encoders. I haven't yet seen a case where there is a mismatch.

`decodeKeys` tries up to 3 variations of the case. Fixes #39